src/config.rs

```diff
@@ -101,69 +101,118 @@ pub struct Provider {
 }
 
 
-pub struct Config {
+pub struct Builder {
```

Comment:
Maybe `ConfigBuilder`? Would mirror `ProviderBuilder` and friends down the road.